File: 08_HCF/HCF.c
#include<stdio.h>

int HCF(int a, int b){
    if(a==0){
        return b;
    }else{
        return HCF(b%a, a);
    }
}

int main()
{
    int num1, num2, hcf;
    printf("First number: ");
    scanf("%d", &num1);

    printf("Second number: ");
    scanf("%d", &num2);

    printf("%d %d", num1, num2);

    hcf = HCF(num1, num2);

    printf("Hcf of given no is: %d", hcf);
    return 0;
}

File: 08_HCF/hcf.py
# Method_1 Using Recursion

def HCForGCD(a, b):

    if(b == 0):
        return a
    else:
        return HCForGCD(b, a%b)

num1, num2 = int(input("First number: ")), int(input("Second number: "))
print("HCF of given number is: ", HCForGCD(num1, num2))


# Method_2 Using Euclidean Algorithm

def HCF(a, b):

    if(a == 0):
        return b
   
    return HCF(b%a, a)

num3, num4 = int(input("First number: ")), int(input("Second number: "))
print("HCF of given number is: ", HCF(num3, num4))